### ISSUE TITLE
Galaxy Skybox

### ISSUE THREAD

**Issue by DerekRies** [DerekRies/Helios#6]
Add a galactic skybox to the current scene.

**Comment by rze89** [comment]
I'll be implementing this as well.

**Comment by rze89** [comment]
Currently haven't been able to find a skybox that we could use that wouldn't be stretched in order to fit. I'm going to continue looking for a suitable image but in the mean time start creating my own. I'll update as soon as some good progress is made.

**Comment by DerekRies** [comment]
Absolute bare minimum for a skybox texture should be 1024x1024. Ideally we would use something high quality like 4096x4096.

Another option is to procedurally generate the skybox. Technology you'd use for this would be the Canvas. An algorithm might look something like this:

1. Create a separate canvas context to render the skybox to.
2. Generate a bunch of small dim stars with random distributions.
3. Use some noise function to generate nebulas, gaseous material.
4. Generate larger brighter stars of varying sizes and intensity.
5. Generate some random color and shift the background hue to that.

Super high-level overview but you get the idea. Let me know what direction you're leaning and I can help you.

**Comment by rze89** [comment]
I actually went ahead and used a tool called SpaceScape that took care of creating a skybox. 4096x4096 seemed a little much so I used a 2048x2048. However rather than one large image, I had to break up the image into 6 separate files(front,back,left,right,up,down). I don't see this as being an issue, since I can "paste" them together in a sense. I'm currently working on implementing the skybox into Helios. I did not originally imagine that implementing the skybox would be as tricky as it has been, however I'm enjoying the learning process.
